src/tests/base_test.py

```diff
@@ -78,6 +82,31 @@ def update_parameterized(cls, request: pytest.FixtureRequest, config: BaseConfig
                 else:
                     raise AttributeError(f"No attribute name {fixture_name} in {config_type} object type")
 
+    @pytest.fixture
+    def k8s_minikube(self, request: pytest.FixtureRequest):
+        """k8 minikube wrapper client  object fixture passed to test function
+        This function fixture called by testcase, we initialize k8client
+        Storing configmap before test begins -> for later restoring
+        When test func done, check if configmap was changed and restore config.
+        :return:
+        """
+        log.debug(f"--- SETUP --- Creating k8s_minikube for test: {request.node.name}\n")
+        k8s_client_ = kubernetes.dynamic.DynamicClient(
+            kubernetes.client.api_client.ApiClient(
+                configuration=kubernetes.config.load_kube_config(),
+            ),
+        )
```

Comment:
Maybe initialize it inside the controller constructor? It's more controller specific logic that is used only inside the controller.

Reply:
ack